### PR TITLE
improvement-NXP-26126-rework-blob-enricher

### PR DESCRIPTION
**Pull request by ataillefer**
T&P: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-ataillefer/137/

### CODE REVIEW

**Comment by efge** [comment]
👍

File: nuxeo-wopi/src/test/java/org/nuxeo/wopi/TestWOPIJsonEnricher.java [hunk 1]
```diff
@@ -147,16 +140,19 @@ public void testMultipleBlobsWOPIJsonEnricher() throws Exception {
...
-            JsonAssert json = jsonAssert(doc, CtxBuilder.enrichDoc(WOPIJsonEnricher.NAME).get());
+            JsonAssert json = jsonAssert(doc, getRenderignContext());
 
             File file = FileUtils.getResourceFileFromContext("json/testMultipleBlobsWOPIJsonEnricher.json");
             String expected = readFile(file, toReplace);
             JSONAssert.assertEquals(expected, json.toString(), false);
         }
     }
 
+    protected RenderingContext getRenderignContext() {
```
Comment (nelsonsilva) [inline nuxeo-wopi/src/test/java/org/nuxeo/wopi/TestWOPIJsonEnricher.java]: nit: typo

**Comment by nelsonsilva** [comment]
👍

File: nuxeo-wopi-web-ui/src/main/resources/web/nuxeo.war/ui/nuxeo-wopi/nuxeo-wopi.html [hunk 2]
```diff
@@ -19,10 +19,6 @@
 
 <nuxeo-slot-content name="wopiLink" slot="BLOB_ACTIONS">
   <template>
-    <nuxeo-filter document="[[document]]" expression="!!document.contextParameters.wopi">
-      <template>
-        <nuxeo-wopi-link document="[[document]]" xpath="[[xpath]]"></nuxeo-wopi-link>
-      </template>
-    </nuxeo-filter>
+    <nuxeo-wopi-link document="[[document]]" blob="[[blob]]"></nuxeo-wopi-link>
```
Comment (troger) [inline nuxeo-wopi-web-ui/src/main/resources/web/nuxeo.war/ui/nuxeo-wopi/nuxeo-wopi.html]: Shouldn't we keep the filter with `expression="!!blob.wopi"`?

File: nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIJsonEnricher.java [hunk 3]
```diff
@@ -16,86 +16,109 @@
...
-            jg.writeEndObject();
+    public void write(JsonGenerator jg, BlobProperty blobProperty) throws IOException {
+        Blob blob = (Blob) blobProperty.getValue();
+        WOPIBlobInfo info = Framework.getService(WOPIService.class).getWOPIBlobInfo(blob);
+        if (info == null) {
+            return;
+        }
+
+        DocumentModel doc = ctx.getParameter(DocumentModelJsonWriter.ENTITY_TYPE);
+        if (doc == null) {
```
Comment (troger) [inline nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIJsonEnricher.java]: I would move up this check, as it seems "less costly".

**Comment by troger** [comment]
:+1:

File: nuxeo-wopi-web-ui/src/main/resources/web/nuxeo.war/ui/nuxeo-wopi/nuxeo-wopi.html (hunk 2 above)
Comment (guirenard) [reply nuxeo-wopi-web-ui/src/main/resources/web/nuxeo.war/ui/nuxeo-wopi/nuxeo-wopi.html]: Unfortunately, `nuxeo-filter` does not take a blob (yet?) so you can't do `<nuxeo-filter blob="[[blob]]" expression="!!blob.wopi">`

**Pull request merged**